packages/library-base/src/commonMain/kotlin/io/realm/kotlin/MutableRealm.kt

```diff
@@ -126,4 +126,23 @@ public interface MutableRealm : TypedRealm {
      * @throws IllegalArgumentException if the object is invalid, frozen or not managed by Realm.
      */
     public fun delete(deleteable: Deleteable)
+
+    /**
+     * Deletes all objects from this Realm.
+     *
+     * @throws IllegalStateException if the Realm is closed.
+     */
+    public fun deleteAll()
+
+    /**
+     * Deletes all objects of the specified class from the Realm.
+     *
+     * @param KClass the class whose objects should be removed.
+     * @throws IllegalStateException if the Realm is closed.
+     */
+    public fun delete(schemaClass: KClass<out BaseRealmObject>)
+}
+
```

Comment:
```suggestion
    /**
     * Deletes all objects of the specified class from the Realm.
     * 
     * Reified convenience wrapper of [MutableRealm.delete].
     */
```
See how other reified extension functions are documented, for example
```
/**
 * Returns a [RealmQuery] matching the predicate represented by [query].
 *
 * Reified convenience wrapper of [TypedRealm.query].
 *
 * @param query the Realm Query Language predicate to append.
 * @param args Realm values for the predicate.
 */
public inline fun <reified T : BaseRealmObject> TypedRealm.query(
    query: String = "TRUEPREDICATE",
    vararg args: Any?
): RealmQuery<T> = query(T::class, query, *args)
```